File: 4 Median of Two Sorted Arrays/C/solution.c
#include "solution.h"

double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int mergedSize = nums1Size + nums2Size;
    int merged[mergedSize];

    int j = 0;
    for (int i = 0; i < nums1Size; i++) {
        merged[j++] = nums1[i]; 
    }
    for (int i = 0; i < nums2Size; i++) {
        merged[j++] = nums2[i];
    }

    qsort(merged, mergedSize, sizeof(int), compare);

    if (mergedSize % 2 == 1) {
        int index = mergedSize / 2;
        return merged[index];
    }
    
    int index1 = mergedSize / 2 - 1;
    int index2 = mergedSize / 2;
    return (double) (merged[index1] + merged[index2]) / 2;
}

int compare(const void* element1, const void* element2) {
    int num1 = *((int*)element1);
    int num2 = *((int*)element2);

    if (num1 > num2) {
        return  1;
    }
    if (num1 < num2) {
        return -1;
    }
    return 0;
}

File: 4 Median of Two Sorted Arrays/Python/solution.py
from typing import List, Self


class Solution:
    def findMedianSortedArrays(self: Self, nums1: List[int], nums2: List[int]) -> float:
        merged: List[int] = []
        for num in nums1:
            merged.append(num)
        for num in nums2:
            merged.append(num)
        lenMerged: int = len(merged)
        merged.sort()
        
        if lenMerged % 2 == 1:
            return merged[lenMerged // 2]
        
        return (merged[lenMerged // 2 - 1] + merged[lenMerged // 2]) / 2
    

if __name__ == "__main__":
    s: Solution = Solution()
    """_summary_
        Input: nums1 = [1,3], nums2 = [2]
        Output: 2.00000
        Explanation: merged array = [1,2,3] and median is 2.
    """
    print(s.findMedianSortedArrays([1,3], [2]))

    """_summary_
        Input: nums1 = [1,2], nums2 = [3,4]
        Output: 2.50000
        Explanation: merged array = [1,2,3,4] and median is (2 + 3) / 2 = 2.5.
    """
    print(s.findMedianSortedArrays([1,2], [3,4]))